compiler/rustc_next_trait_solver/src/delegate.rs

```diff
@@ -29,10 +29,10 @@ pub trait SolverDelegate: Deref<Target = <Self as SolverDelegate>::Infcx> + Size
     // FIXME: Uplift the leak check into this crate.
     fn leak_check(&self, max_input_universe: ty::UniverseIndex) -> Result<(), NoSolution>;
 
-    fn try_const_eval_resolve(
+    fn evaluate_const(
         &self,
         param_env: <Self::Interner as Interner>::ParamEnv,
-        unevaluated: ty::UnevaluatedConst<Self::Interner>,
```

Comment:
We use a `Const` not a `UnevaluatedConst` as `traits::evaluate_const` accepts a `Const` not a `UnevaluatedConst` so that the normalization routines can just call it on their `Const` instead of having to match on the `ConstKind`. It wouldn't surprise me if we changed this when proper associated const normalization is implemented

Reply:
When we implement "proper" normalization of associated constants I expect us to want to start matching on the `ConstKind` in the normalization folders, at which point `traits::evaluate_const` should probably just take `ty::UnevaluatedConst`, at which point yeah we probably change this back to `ty::UnevaluatedConst`.

Reply:
👍